### PR TITLE
Replace deprecated react hls player

### PR DESCRIPTION
**Pull request by seyeong-han**
### Parent Issue
[2. react-hls-player dependency conflict issue with react@18.0.0](#68)  

I got the [react-hls-player](https://github.com/devcshort/react-hls) dependency conflict issue and found it has been stopped from being maintained since 2021.

## Fix
- Replaced the `react-hls-player` with `react-player@2.16.0`.
- Fixed `useVideoElementEffectProps` types for react-player

## Result
image

### CODE REVIEW

**Comment by DmitriiTsy** [comment]
Good idea @seyeong-han  Switching to react-player is definitely the way to go since it gives us out of the box support for different & multiple media sources like YouTube, Vimeo, and etc , which is perfect for scaling up . Plus, with its built-in controls and playback events, we’ll have more flexibility and control do develop better features in the future.  


I think overall changes is looks good, thank you for contributing @seyeong-han appreciate it 

I have a couple important things to mention. We were using "poster" as a way to show a thumbnail of the video while videos is loading (instead of a black screen or smith like this) I know that in this current library we have a specific tool to do the same called "light" 

The other thing is this thumbnail  (poster/light) is relying a lot on the dimensions and they should be strict like 850px for width and 560px for height (we can use our previous ones as well )

**Review by DmitriiTsy — CHANGES_REQUESTED** [review]

File: frontend/src/components/Modal/ModalCentral.tsx [hunk 1]
```diff
@@ -99,14 +99,19 @@ arrayOfChoosedElements?.[chosenIndex as number]?.thumbnail_url as string :
             />
           </div>
           <div>
-            <ReactHlsPlayer
-              poster={fallbackImage}
```
[the remark this answers is not in the document]
Comment (seyeong-han) [reply frontend/src/components/Modal/ModalCentral.tsx:103]: I fixed it in my new commit!

File: frontend/src/components/Modal/ModalCentral.tsx [hunk 2]
```diff
@@ -99,14 +99,19 @@ arrayOfChoosedElements?.[chosenIndex as number]?.thumbnail_url as string :
...
-              src={videoUrl}
-              width={'854px'}
+            <ReactPlayer
+              ref={videoRef}
+              url={videoUrl}
+              width="100%"
```
[the remark this answers is not in the document]
Comment (seyeong-han) [reply frontend/src/components/Modal/ModalCentral.tsx]: If I set the width to 850px and the height to 560px, the modal does not align properly with the video's dimensions. Even though it doesn't fit, I followed your suggested size adjustments.

image

[the remark this answers is not in the document]
Comment (seyeong-han) [reply frontend/src/components/Modal/ModalCentral.tsx]: @DmitriiTsy I’ve updated the code based on your suggestions!

**Comment by DmitriiTsy** [comment]
Since the last time we checked this PR, we have implemented a lot of UI changes that may affect some of the changes in this PR. Therefore, this PR can be closed for now @seyeong-han

**Pull request closed without merging**